src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -383,3 +385,238 @@ ShaderNetworkPtr ShaderNetworkAlgo::convertObjectVector( const ObjectVector *net
+	auto &values = valuesData->writable();
+	values.reserve( spline.points.size() );
+
+	if( duplicateEndPoints && spline.points.size() )
+	{
+		positions.push_back( spline.points.begin()->first );
+		values.push_back( spline.points.begin()->second );
+	}
+	for( typename Spline::PointContainer::const_iterator it = spline.points.begin(), eIt = spline.points.end(); it != eIt; ++it )
+	{
+		positions.push_back( it->first );
+		values.push_back( it->second );
+	}
+	if( duplicateEndPoints && spline.points.size() )
+	{
+		positions.push_back( spline.points.rbegin()->first );
+		values.push_back( spline.points.rbegin()->second );
+	}
+
+	newParameters[ name.string() + "Positions" ] = positionsData;
+	newParameters[ name.string() + "Values" ] = valuesData;
+	newParameters[ name.string() + "Basis" ] = new StringData( basis );
+}
+
+template<typename SplineData>
+IECore::DataPtr loadSpline(
+	const StringData *basisData,
+	const IECore::TypedData< std::vector< typename SplineData::ValueType::XType > > *positionsData,
+	const IECore::TypedData< std::vector< typename SplineData::ValueType::YType > > *valuesData
+)
+{
+	typename SplineData::Ptr resultData = new SplineData();
+	auto &result = resultData->writable();
+
+	bool unduplicateEndPoints = false;
+
+	const std::string &basis = basisData->readable();
+	if( basis == "bezier" )
+	{
+		result.basis = SplineData::ValueType::Basis::bezier();
+	}
+	if( basis == "bspline" )
+	{
+		result.basis = SplineData::ValueType::Basis::bSpline();
+	}
+	else if( basis == "linear" )
+	{
+		unduplicateEndPoints = true;
+		result.basis = SplineData::ValueType::Basis::linear();
+	}
+	else
+	{
+		result.basis = SplineData::ValueType::Basis::catmullRom();
+	}
+
+	const auto &positions = positionsData->readable();
+	const auto &values = valuesData->readable();
+
+	size_t n = std::min( positions.size(), values.size() );
+	for( size_t i = 0; i < n; ++i )
+	{
+		if( unduplicateEndPoints && ( i == 0 || i == n - 1 ) )
+		{
+			continue;
+		}
+
+		result.points.insert( typename SplineData::ValueType::Point( positions[i], values[i] ) );
+	}
+
+	return resultData;	
+}
+
+} // namespace
+
+IECore::ConstCompoundDataPtr ShaderNetworkAlgo::collapseSplineParameters( IECore::ConstCompoundDataPtr parameters )
+{
+	CompoundDataPtr newParametersData;
+
+	const auto &parms = parameters->readable();
+	for( const auto &maybeBasis : parameters->readable() )
+	{
+		if( !boost::ends_with( maybeBasis.first.string(), "Basis" ) )
+		{
+			continue;
+		}
+		const StringData *basis = runTimeCast<const StringData>( maybeBasis.second.get() );
+		if( !basis )
+		{
+			continue;
+		}
+
+
+		std::string prefix = maybeBasis.first.string().substr( 0, maybeBasis.first.string().size() - 5 );
+		IECore::InternedString positionsName = prefix + "Positions";
+		const auto positionsIter = parms.find( positionsName );
+		const FloatVectorData *floatPositions = nullptr;
+		const DoubleVectorData *doublePositions = nullptr;
```

Comment:
Do we need to support doubles here? I'm not aware of anywhere else they're supported.

Reply:
I dunno, I just saw that Splinedd exists, and I was replacing prepareSplineCVsForOSL, which was templated to handle anything, and I figured I should cover my bases.  Happy to remove this support if you want.

Reply:
Yeah, let's ditch it. No point maintaining code that won't be used.